docs/docs/getting-started/developing-on-hydra.md

```diff
@@ -0,0 +1,89 @@
+This off-chain code will usually be exposed through an HTTP interface that frontends can access to drive interactions with the application forward.
+Off-chain code is usually written in the Contract monad, which is used to build transactions and interact with the Cardano blockchain in a controlled way.
+
+When it comes to hydra, the developer will not have access to similar tools. The interface to speak to a hydra node is a websocket. Developers must build applications that connect to these websockets and can send and receive events through them.
+
+The Hydra node will emit several commands through this websocket: the API is defined [here](https://hydra.family/head-protocol/api-reference).
+The API reference is divided into two sections: Pub and Sub.
+The first one describes the commands that a hydra node will accept, while the second one describes the events that will be emitted through the websocket.
+
+An important event developers will be interested in is `SnaphotConfirmed`. This event is emitted once all the hydra nodes that are partaking in the head, have seen and signed a transaction submitted by one of them.
+
+Once a transaction lands inside a snapshot, it is confirmed and there are no chances for it to be rolled back. This is an important distinction from L1: in that case, when a node sees a transaction there is still a chance that it might be rolled back (where the probability of this goes to 0 as more blocks are added), in the hydra world, the only way to make a snapshot is to have every single node in the head approve of it.
+This is a synchronous process and requires all the head participants to be online all the time, the tradeoff is that there is no uncertainty or possibility of rollbacks, the moment the snapshot is confirmed there is no going back.
+
+For the Pub side of the API, an important endpoint will be `NewTx`. This is used to submit a transaction to the node. There are a couple of ways the transaction can be submitted, but the simplest method we found is to simply serialise the transaction built with `Cardano.Api` to CBOR and submit that through a websocket.
+
+## Testing
+
+Testing inside the hydra head can be quite different than for normal DApps, the reason is you will not have access to EmulatorTrace or an easy way to spin up a dev environment to test.
+
+However it is possible to spin up a cardano-node, and a cluster of hydra-nodes programmatically and run tests directly on that.
+The module hydra-cluster provides functions to spin up the set of required nodes and to run assertions on their output.
+
+A couple of steps are required to set up the initial environment:
+
+- We need to spin up at least one Cardano BFT node. This can be done with `withBFTNode` which takes the cardano node configuration and a callback that has access to an instance of the running node. The node parameters can be generated by a helper also exported by hydra-cluster: `newNodeConfig`.
```

Comment:
Quite some things will / have been renamed in #430. I'll suggest the new names:

```suggestion
- We want to spin up a local Cardano devnet comprised by a single block producer. This can be done with `withCardanoDevnet` which takes the a working directory where blocks and logs will be stored, as well as a callback that gets access to the `RunningNode`. Note that the devnet will use credentials and configuration from the `hydra-cluster/config/`.
```